### ISSUE TITLE
More Shapes/Primitives!

### ISSUE THREAD

**Issue by cqann** [Canvim/PyPen#13]
I suggest that more shapes should be added to PyPen! 
More specifically these primitives should be added:
* Triangle
* Line
* Custom Shape (with begin_shape and end_shape)
* potentially square, equilateral_triangle, point (just a pixel) and curve

**Comment by EriKWDev** [comment]
N-gon, centered_rectangle and centered_square :3 (if we don't implement them in a different way)

**Comment by cqann** [comment]
I agree that shapes such as rectangles should be able to be centered, but I don't believe that they should have their own primitives. Instead I'm suggesting that we maybe implement a rectangle_mode, either a function or variable. This would contain info which would be checked in the rectangle primitive, and  translation would then be carried out there.

**Comment by EriKWDev** [comment]
Different modes seems like a good idea :) settings.rectangle_mode = "normal" | "center"/"middle" could do that.
We should probably, speaking of modes, also have settings.angle_mode = "degrees" | "radians"

**Comment by EriKWDev** [comment]
> Instead I'm suggesting that we maybe implement a rectangle_mode, either a function or variable

see #17
